deepchecks/tabular/dataset.py

```diff
@@ -302,22 +277,30 @@ def __init__(
             else:
                 self._data[self._datetime_name] = pd.to_datetime(self._data[self._datetime_name], **self._datetime_args)
 
-        if label_type and self._label_name:
-            if label_type == 'regression_label':
-                self._label_type = TaskType.REGRESSION
-            elif label_type == 'classification_label':
-                if self.data[self._label_name].nunique() > 2:
-                    self._label_type = TaskType.MULTICLASS
-                else:
-                    self._label_type = TaskType.BINARY
-            else:
-                get_logger().warning('Label type %s is not valid, auto inferring label type.'
-                                     ' Possible values are regression_label or classification_label.', label_type)
-        if self._label_name and not hasattr(self, "label_type"):
-            self._label_type = self._infer_label_type(self.data[self._label_name])
-        elif not hasattr(self, "label_type"):
+        if label_type in ['classification_label', 'regression_label']:
+            warnings.warn(f'{label_type} value for label type is deprecated, allowed task types are multiclass,'
+                          f' binary and regression.', DeprecationWarning, stacklevel=2)
+            self._label_type = TaskType.REGRESSION if label_type == 'regression_label' else TaskType.MULTICLASS
+        elif label_type in [task.value for task in TaskType]:
+            self._label_type = TaskType(label_type)
+        elif label_type is not None:
+            raise DeepchecksValueError(f'allowed value for label type are {[task.value for task in TaskType]},'
+                                       f' received {label_type}.')
+        else:
             self._label_type = None
 
+        if label_classes is not None:
+            if not isinstance(label_classes, t.List) or len(set(label_classes)) != len(label_classes) or \
+                    any(isinstance(x, t.Sequence) and not isinstance(x, str) for x in label_classes):
+                raise DeepchecksValueError('label_classes must be a flat list of unique values.')
+            if label is not None and not set(self.label_col).issubset(set(label_classes)):
+                raise DeepchecksValueError('label_classes does not contain all labels found in label column.')
+            if label_classes != sorted(label_classes):
+                raise DeepchecksValueError('label_classes order must correspond to the required order of the model\'s '
+                                           'probabilities per class (alphanumeric order). For additional details see'
+                                           'https://docs.deepchecks.com/dev/user-guide/tabular/supported_models.html')
```

Comment:
Use `doclink` to replace the "dev" version with the current version of the installed package.